pg_search/src/schema/document.rs

```diff
@@ -40,114 +36,3 @@ impl From<SearchDocument> for TantivyDocument {
         value.doc
     }
 }
-
-fn serialize_document<S>(doc: &TantivyDocument, serializer: S) -> Result<S::Ok, S::Error>
-where
-    S: Serializer,
-{
-    let mut field_values: Vec<(String, String, OwnedValue)> = vec![];
-    for (field, value) in doc.field_values() {
-        let field_string = serde_json::to_string(&field).map_err(serde::ser::Error::custom)?;
-        // We have to store the type for the following because positive i64s and dates get automatically deserialized as u64s
-        let owened_value = tantivy::schema::document::OwnedValue::from(value);
-        let (value_type, value) = match owened_value {
-            tantivy::schema::document::OwnedValue::I64(i64) => (
-                "i64".into(),
-                tantivy::schema::document::OwnedValue::Str(
-                    serde_json::to_string(&i64).map_err(serde::ser::Error::custom)?,
-                ),
-            ),
-            tantivy::schema::document::OwnedValue::Date(date) => (
-                "date".into(),
-                tantivy::schema::document::OwnedValue::Str(
-                    serde_json::to_string(&date).map_err(serde::ser::Error::custom)?,
-                ),
-            ),
-            val => ("other".into(), val.clone()),
-        };
-        field_values.push((field_string, value_type, value.clone()));
-    }
-
-    let doc_string = serde_json::to_string(&field_values).map_err(serde::ser::Error::custom)?;
-
-    serializer.serialize_str(&doc_string)
-}
-
-fn deserialize_document<'de, D>(deserializer: D) -> Result<TantivyDocument, D::Error>
```

Comment:
Nice to see this code go. Is there a reason why we were doing custom serialize/deserialize in the first place?

Reply:
The background worker index writer, I believe.

Reply:
Yep that's right, we needed it for IPC to the background process, and we used bincode for an efficient representation. The feature we needed didn't support `derive`, so here we were!